src/navi/core.clj

```diff
@@ -27,6 +31,18 @@
     (contains? m k)
     (update-in [k] #(into [:map] %))))
 
+(defn update-kvs
+  "Update a map using `key-fn` and `val-fn`.
+  Sort of like composing `update-keys` and `update-vals`.
+  Unlike `update-keys` or `update-vals`, preserve nils."
+  [m key-fn val-fn]
+  (if (nil? m)
+    nil
+    (reduce-kv (fn kv-mapper [m k v]
```

Comment:
This is neat! I hammocked a bit more and came up with this:
```clojure
(into {} (for [[k v] m] [(key-fn k) (val-fn v)])
```
feels simpler to me? lemme know what you think? `reduce-kv` is indeed more relevant but also feels a bit overkill? not sure, im open to thoughts!

Reply:
Also (when m ...) can be used here instead of (if (nil? m) ...).

Reply:
Thanks for the additional suggestion.

My personal feeling is that `reduce-kv` is the more appropriate function in this situation, and reflects the intent more.  It is also faster (approximately 80 uS vs 130 uS), most likely because of the intermediate sequence generated by the `for`.  My feeling is to leave it as is (using `reduce-kv`).  However, I am happy to make the change to use the `for` version if you'd prefer.  Let me know.

FWIW, here is the transcript of the benchmarking session.

```clojure
navi.core> (require '[criterium.core :as c])
nil
navi.core> (def x (into {} (for [i (range 1000)] [(str "k" i) i])))
#'navi.core/x
navi.core> (defn lispyclouds-update-kvs
             "Update a map using `key-fn` and `val-fn`.
  Sort of like composing `update-keys` and `update-vals`.
  Unlike `update-keys` or `update-vals`, preserve nils."
             [m key-fn val-fn]
             (if (nil? m)
               nil
               (into {} (for [[k v] m] [(key-fn k) (val-fn v)]))))
#'navi.core/lispyclouds-update-kvs
navi.core> (c/quick-bench (update-kvs x identity identity))
Evaluation count : 7560 in 6 samples of 1260 calls.
             Execution time mean : 78.108056 µs
    Execution time std-deviation : 930.911015 ns
   Execution time lower quantile : 77.313815 µs ( 2.5%)
   Execution time upper quantile : 79.620446 µs (97.5%)
                   Overhead used : 1.594769 ns

Found 1 outliers in 6 samples (16.6667 %)
	low-severe	 1 (16.6667 %)
 Variance from outliers : 13.8889 % Variance is moderately inflated by outliers
nil
navi.core> (c/quick-bench (update-kvs x identity identity))
Evaluation count : 6024 in 6 samples of 1004 calls.
             Execution time mean : 87.853485 µs
    Execution time std-deviation : 14.268741 µs
   Execution time lower quantile : 79.987076 µs ( 2.5%)
   Execution time upper quantile : 112.570970 µs (97.5%)
                   Overhead used : 1.594769 ns

Found 1 outliers in 6 samples (16.6667 %)
	low-severe	 1 (16.6667 %)
 Variance from outliers : 47.5319 % Variance is moderately inflated by outliers
nil
navi.core> (c/quick-bench (lispyclouds-update-kvs x identity identity))
Evaluation count : 4938 in 6 samples of 823 calls.
             Execution time mean : 129.911469 µs
    Execution time std-deviation : 8.714619 µs
   Execution time lower quantile : 120.260354 µs ( 2.5%)
   Execution time upper quantile : 141.261373 µs (97.5%)
                   Overhead used : 1.594769 ns
nil
navi.core> (c/quick-bench (lispyclouds-update-kvs x identity identity))
Evaluation count : 4824 in 6 samples of 804 calls.
             Execution time mean : 130.544097 µs
    Execution time std-deviation : 5.086537 µs
   Execution time lower quantile : 125.795358 µs ( 2.5%)
   Execution time upper quantile : 137.748456 µs (97.5%)
                   Overhead used : 1.594769 ns
nil
navi.core> 
```

Reply:
Agree on using the `when`.

Reply:
Great work on the benchmarking and agree on keeping reduce-kv. Will merge as soon as I get to a computer.